var/spack/repos/builtin/packages/catalyst/package.py

```diff
@@ -35,12 +36,12 @@ class Catalyst(CMakePackage):
     variant('extras', default=False, description='Enable Extras support')
     variant('rendering', default=False, description='Enable VTK Rendering support')
     variant('osmesa', default=True, description='Use offscreen rendering')
-    conflicts('+osmesa', when='~rendering')
+    conflicts('+osmesa', when='+rendering')
```

Comment:
Aside from being out of scope for this python change, the original conflict specification is correct.  `+rendering` will enable or disable any rendering support.  `+osmesa` is used to set which rendering back-end get's enabled.

Reply:
I see. Indeed, you are right, the original spec is correct. The problem was with the default parameters: osmesa=on, rendering=off.